wgengine/magicsock/magicsock.go

```diff
@@ -1294,6 +1294,15 @@ func (c *Conn) sendUDP(ipp netip.AddrPort, b []byte, isDisco bool) (sent bool, e
 // any conditionals are met.
 func (c *Conn) maybeRebindOnError(os string, err error) bool {
 	switch {
+	case errors.Is(err, syscall.EPIPE) || errors.Is(err, syscall.ENOTCONN):
+		// EPIPE/ENOTCONN are common errors when a send fails due to a closed
+		// socket. There is some platform and version inconsistency in which
+		// error is returned, but the meaning is the same.
+		why := "broken-pipe-rebind"
+		c.logf("magicsock: performing %q", why)
+		c.Rebind()
+		go c.ReSTUN(why)
+		return true
 	case errors.Is(err, syscall.EPERM):
```

Comment:
This method returning a bool is a little weird (which only tests care about). And repeating its body twice.

Maybe this can be broken in half: maybeRebindOnError which does the log + rebind and doesn't return anything, then the policy half that returns the string reason (or empty string if none) and then `TestMaybeRebindOnError` can test that side-effect-free method instead?

Reply:
Followed up in #14552